segmentstore/server/host/src/main/java/io/pravega/service/server/host/stat/AutoScaleProcessor.java

```diff
@@ -185,18 +186,29 @@ private void triggerScaleDown(String streamSegmentName, boolean silent) {
 
     private CompletableFuture<Void> writeRequest(ScaleEvent event) {
         CompletableFuture<Void> result = new CompletableFuture<>();
+        CompletableFuture<Void> writeComplete = new CompletableFuture<>();
         try {
-            CompletableFuture.runAsync(() -> {
+            AckFuture ackFuture = writer.get().writeEvent(event.getKey(), event);
+            ackFuture.addListener(() -> writeComplete.complete(null), executor);
+            writeComplete.thenAcceptAsync((Void v) -> {
                 try {
-                    writer.get().writeEvent(event.getKey(), event).get();
-                    result.complete(null);
-                } catch (InterruptedException | ExecutionException e) {
-                    log.error("error sending request to requeststream {}", e);
+                    ackFuture.get();
+                } catch (ExecutionException | InterruptedException e) {
+                    log.error("Sending scale event failed {}/{}/{}", event.getScope(), event.getStream(), event.getSegmentNumber());
+                    result.completeExceptionally(ExceptionHelpers.getRealException(e));
+                }
+            }).whenComplete((r, e) -> {
+                if (e != null) {
+                    log.error("Sending scale event failed {}/{}/{}", event.getScope(), event.getStream(), event.getSegmentNumber());
```

Comment:
This is redundant with the code a few lines above. I think it would simplify things if you did a thenAccept() which did not have any error handling followed by a .exceptionally which does.